core/unit_test/TestViewAPI.hpp

```diff
@@ -1340,6 +1340,38 @@ class TestViewAPI {
     ASSERT_EQ(dz.data(), nullptr);
   }
 
+  struct test_refcount_poison_copy_functor {
+    using view_type = Kokkos::View<double *>;
+    explicit test_refcount_poison_copy_functor(view_type v) : view(v) {}
+
+    test_refcount_poison_copy_functor(
+        const test_refcount_poison_copy_functor &other)
+        : view(other.view) {
+      throw std::bad_alloc();
+    }
+
+    KOKKOS_INLINE_FUNCTION void operator()(int i) const { view(i) = i; }
```

Comment:
Do you need to write to the `view` here (to check that capturing the view by copy works as intended) or could this also be a no-op?